.github/workflows/release.yaml

```diff
@@ -1,16 +1,22 @@
 name: Release
 on:
   push:
-    tags:
-      - 'v*'
+    branches:
+      - release
 jobs:
-  deploy:
-    name: Create GitHub Release
+  release:
     runs-on: ubuntu-20.04
+    environment: NPM
     steps:
       - uses: actions/checkout@v3
-      - uses: ncipollo/release-action@v1
         with:
-          artifacts: 'CHANGELOG.md,LICENSE,README.md'
-          generateReleaseNotes: true
-          token: ${{ secrets.GITHUB_TOKEN }}
+          persist-credentials: false
+      - uses: actions/setup-node@v3
+        with:
+          cache: 'npm'
+          node-version: 16
+      - run: npm ci
+      - env:
+          GITHUB_TOKEN: ${{ secrets.ACTION_TOKEN }}
```

Comment:
We should be able to just use `${{ secrets.GITHUB_TOKEN }}` here? And then we can remove the `environment: NPM`?

Reply:
no, because the `release` workflow is based on tags. when using `GITHUB_TOKEN`, the tag push will not triger any other actions. it must be a custom token.

either way, you still need `NPM` for the npmjs.org credentials (in `release`).

Reply:
or we could go for [`@semantic-release/npm`](https://www.npmjs.com/package/@semantic-release/npm) for the release. I've never used that extension specifically, but I assume it would work.

Reply:
> no, because the `release` workflow is based on tags. when using `GITHUB_TOKEN`, the tag push will not triger any other actions. it must be a custom token.
> 
> either way, you still need `NPM` for the npmjs.org credentials (in `release`).

Ah right, I forgot that part.

Can you change environment to `GITHUB` and add this action before the step that requires the token:

```yaml
      - uses: tibdex/github-app-token@v1.6.0
        id: generate-token
        with:
          app_id: ${{ secrets.APP_ID }}
          private_key: ${{ secrets.APP_PK }}
```

And then use the token like this:

```yaml
GITHUB_TOKEN: ${{ steps.generate-token.outputs.token }}
````

Reply:
that's a cool solution. I can do that :)

Reply:
Not sure, the release event may also be limited by the same problem.
Will see, let's give it a try.